geonode/harvesting/resourcedescriptor.py

```diff
@@ -88,3 +89,72 @@ class RecordDescription:
     identification: RecordIdentification
     distribution: RecordDistribution
     data_quality: str
+
+
+# TODO:
+#  Not sure this is needed, because layers does not have information for RecordDescriptionContact
+#  And some of it
+@dataclasses.dataclass()
+class ResourceBaseData:
+    uuid: uuid.UUID
+    abstract: typing.Optional[str]
+    alternate: typing.Optional[str]
+    attribution: typing.Optional[str]
+    bbox_polygon: typing.Optional[geos.Polygon]
+    constraints_other: typing.Optional[str]
+    created: typing.Optional[dt.datetime]
+    data_quality_statement: typing.Optional[str]
+    date: typing.Optional[dt.datetime]
+    date_type: typing.Optional[str]
+    doi: typing.Optional[str]
+    edition: typing.Optional[str]
+    language: typing.Optional[str]
+    purpose: typing.Optional[str]
+    supplemental_information: typing.Optional[str]
+    temporal_extent_end: typing.Optional[str]
+    temporal_extent_start: typing.Optional[str]
+    title: str
+
+    @abc.abstractmethod
+    def to_json(
+            self
+    ) -> dict:
+        """
+        Return data as json for model defaults
+        """
+
+
+@dataclasses.dataclass()
+class ResourceData:
+    resource_base_data: ResourceBaseData
+
+    def to_json(
+            self
+    ) -> dict:
+        """
+        Return data as json for model defaults
+        """
+        output = self.resource_base_data.__dict__
+        output.update(self.__dict__)
+        del output['resource_base_data']
+        return output
+
+
+@dataclasses.dataclass()
+class LayerData(ResourceData):
+    resource_base_data: ResourceBaseData
+
+
+@dataclasses.dataclass()
+class MapData(ResourceData):
+    resource_base_data: ResourceBaseData
+    zoom: int
+    projection: str
+    center_x: float
+    center_y: float
+    last_modified: typing.Optional[dt.datetime]
+
```

Comment:
It is not clear to me why you would need to add these additional classes. `ResourceBaseData` seems to be repeating some of the stuff that is already in the `RecordIdentification` class. Why did you not reuse and enhance the classes that were already in this module?

The idea is to have a single `ResourceDescriptor` class that can be used to create all resource types (layer, document, etc). This means you need to change some of the current parameters to make them optional and potentially add some free type parameters. Ideally we don't want to keep adding custom classes here for every new resource type.

Reply:
Updated @ricardogsilva